src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertyAccessorSymbol.cs

```diff
@@ -152,8 +152,8 @@ internal override bool IsExpressionBodied
             _isExpressionBodied = true;
 
             // The modifiers for the accessor are the same as the modifiers for the property,
-            // minus the indexer bit
-            var declarationModifiers = propertyModifiers & ~DeclarationModifiers.Indexer;
+            // minus the indexer and readonly bit
+            var declarationModifiers = propertyModifiers & ~(DeclarationModifiers.Indexer | DeclarationModifiers.ReadOnly);
```

Comment:
>propertyModifiers & ~(DeclarationModifiers.Indexer | DeclarationModifiers.ReadOnly) [](start = 39, length = 83)

Consider extracting a `static` helper method for use here and in constructor below.

Reply:
You mean a method like `GetAccessorModifiers(propertyModifiers)`? Then do the special private virtual tweak in the constructor taking AccessorDeclarationSyntax?

---
In reply to: [267972053](https://github.com/dotnet/roslyn/pull/34338#discussion_r267972053) [](ancestors = 267972053)

Reply:
Yes, use `GetAccessorModifiers(propertyModifiers)` in both, and keep the additional tweak in the second constructor.

---
In reply to: [267976876](https://github.com/dotnet/roslyn/pull/34338#discussion_r267976876) [](ancestors = 267976876,267972053)